crates/burn-import/src/burn/node/constant_of_shape.rs

```diff
@@ -0,0 +1,185 @@
+            Self::Float64(val) => quote! { #val },
+            Self::Int32(val) => quote! { #val },
+            Self::Int64(val) => quote! { #val },
+            Self::Bool(val) => quote! { #val },
+        }
+    }
+
+    pub fn from_vec<T: Into<Self> + Copy>(mut source: Vec<T>) -> Self {
+        assert_eq!(
+            source.len(),
+            1,
+            "ConstantOfShape value from a vec needs to have exactly 1 element!"
+        );
+        source.drain(..).next().unwrap().into()
+    }
+}
+
+impl From<f32> for ConstantValue {
+    fn from(value: f32) -> Self {
+        Self::Float32(value)
+    }
+}
+impl From<f64> for ConstantValue {
+    fn from(value: f64) -> Self {
+        Self::Float64(value)
+    }
+}
+impl From<i32> for ConstantValue {
+    fn from(value: i32) -> Self {
+        Self::Int32(value)
+    }
+}
+impl From<i64> for ConstantValue {
+    fn from(value: i64) -> Self {
+        Self::Int64(value)
+    }
+}
+impl From<bool> for ConstantValue {
+    fn from(value: bool) -> Self {
+        Self::Bool(value)
+    }
+}
+
+impl<PS: PrecisionSettings> NodeCodegen<PS> for ConstantOfShapeNode {
+    fn input_types(&self) -> Vec<Type> {
+        vec![self.input.clone()]
+    }
+
+    fn output_types(&self) -> Vec<Type> {
+        vec![self.output.clone()]
+    }
+
+    fn forward(&self, _scope: &mut Scope, _node_position: usize) -> TokenStream {
+        let output = self.output.name();
+        let input = self.input.name();
+        let value = self.value.val_tokens();
+        // Note: in the generated code, self.device is a &module::Ignored<Device>, so to get a &Device, &* is needed:
+        quote! {
+            let #output = Tensor::full(#input.shape(), #value, &*self.device);
+        }
+    }
+
+    fn into_node(self) -> Node<PS> {
+        Node::ConstantOfShape(self)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use burn::record::FullPrecisionSettings;
+
+    use super::*;
+    use crate::burn::{
+        graph::BurnGraph,
+        node::{constant_of_shape::ConstantOfShapeNode, test::assert_tokens},
+        TensorType,
+    };
+
+    #[test]
+    fn test_constant_val() {
+        assert_eq!(ConstantValue::from(1i32), ConstantValue::Int32(1i32));
+        assert_eq!(ConstantValue::from(-1i64), ConstantValue::Int64(-1i64));
+        assert_eq!(ConstantValue::from(0f32), ConstantValue::Float32(0f32));
+        assert_eq!(ConstantValue::from(0f64), ConstantValue::Float64(0f64));
+        assert_eq!(ConstantValue::from(true), ConstantValue::Bool(true));
+        assert_eq!(
+            ConstantValue::from_vec(vec![2i32]),
+            ConstantValue::Int32(2i32)
+        );
+    }
+
+    #[test]
+    fn test_codegen_nodes() {
+        let mut graph = BurnGraph::<FullPrecisionSettings>::default();
+
+        graph.register(ConstantOfShapeNode::new(
+            Type::Tensor(TensorType::new_float("tensor1", 4)),
+            Type::Tensor(TensorType::new_float("tensor2", 4)),
```

Comment:
The input `tensor1` should have rank 1 (not 4).

Reply:
Fixed it by introducing shape type.